kgforge/specializations/models/rdf/service.py

```diff
@@ -138,12 +137,19 @@ def lookup_shape_from_node(self, node: URIRef) -> Shape:
             shape = self._node_shape_cache[node]
         except KeyError as ke:
             raise ValueError(f"Unknown shape node id '{node}': {str(ke)}") from ke
-        if shape:
-            shape_wrapper = ShapeWrapper(self._node_shape_cache[node])
-            if not hasattr(shape_wrapper, "traverse"):
-                shape_wrapper.traverse = types.MethodType(traverse, shape_wrapper)
-            return shape_wrapper
-        return shape
+
+        return ShapeWrapper(shape)
+
+    @property
+    def shapes(self):  # pyshacl implementation returns dict_values (not list). This cannot be pickled.
```

Comment:
maybe `shapes_list` would be even more accurate?

Reply:
This is overriding a property defined by pyshacl, so I have to stick to the same name

Reply:
Ah ok